OVS issues found by static analyzis

cc @vmytnykx  @MariuszX 

Hi!

Please look carefully and fix issues found by static analyzis.
That are not false triggered but a real issue.

Also, please subscribe/inform other concerned persons too.

Thanks.

```
src/utils_ovs.c:1057:7: warning: Use of memory after it is freed
  if (ovs_db_poll_thread_init(pdb) < 0) {
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/utils_ovs.c:1064:3: warning: Use of memory after it is freed
  return pdb;
  ^~~~~~~~~~
2 warnings generated.
  CCLD     ovs_events.la
  CC       src/ovs_stats_la-ovs_stats.lo
  CC       src/ovs_stats_la-utils_ovs.lo

src/utils_ovs.c:1057:7: warning: Use of memory after it is freed
  if (ovs_db_poll_thread_init(pdb) < 0) {
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/utils_ovs.c:1064:3: warning: Use of memory after it is freed
  return pdb;
  ^~~~~~~~~~
```

--- a/src/utils_ovs.c
+++ b/src/utils_ovs.c
@@ -1051,6 +1051,8 @@ ovs_db_t *ovs_db_init(const char *node, const char *service,
     ret = ovs_db_destroy(pdb);
     if (ret > 0)
       goto failure;
+    else
+      return NULL;
   }
 
   /* init polling thread */
@@ -1059,6 +1061,8 @@ ovs_db_t *ovs_db_init(const char *node, const char *service,
     if (ret > 0) {
       ovs_db_event_thread_data_destroy(pdb);
       goto failure;
+    } else {
+      return NULL;
     }
   }
   return pdb;

